### PR TITLE
Add Ability to remove Unused Profiles

### PR DESCRIPTION
**Pull request by techgurupezza**
## Feature Enhancements Added
- #21 

## Bugs Fixed
N/A

## Description of overall changes
Allows the ability to delete a profile from settings view.

### CODE REVIEW

File: app/settings/profile.coffee [hunk 1]
```diff
@@ -16,6 +16,22 @@ class Profile extends React.Component
       onClick: (e) =>
         @props.setActiveProfile(@props.profile.id)
       @textOrInput()
+      React.createElement(
+        'span'
+        className: 'profile-remove'
+        onClick: (e) =>
+          tempState = {}
```
Comment (npezza93) [inline app/settings/profile.coffee]: Set this as a function, `removeProfile`, on Profiles and pass it through the props like with `setActiveProfile`. Should be very similar to `createProfile`.

File: app/settings/profile.coffee [hunk 2]
```diff
@@ -16,6 +16,22 @@ class Profile extends React.Component
...
+
+          Object.assign(tempState, @props.configFile.contents())
+          delete tempState.profiles[@props.profile.id]
+          @props.configFile.write(tempState)
+
+          @props.updateProfileList(Object.keys(Object.assign(
+            tempState, @props.configFile.contents()
+          ).profiles)[0])
+          e.stopPropagation()
+        'x'
```
Comment (npezza93) [inline app/settings/profile.coffee]: Use the same `x` used in the ArchipelagoTab

File: app/settings/settings.css [hunk 3]
```diff
@@ -97,6 +97,14 @@ input,
   font-weight: 400;
   font-family: -apple-system, system-ui, "Segoe UI", Helvetica, Arial, sans-serif, "Apple Color Emoji", "Segoe UI Emoji", "Segoe UI Symbol";
   font-size: 16px;
+  position: relative;
+  z-index: 15;
+}
+
+.profiles-container archipelago-profiles .profile-list archipelago-profile .profile-remove {
```
Comment (npezza93) [inline app/settings/settings.css]: I would prefer this to be on the left side, in the margin and only visible when you hover over the profile.

Comment (ncino-cpezza) [reply app/settings/settings.css]: Moved to left and added hover state @npezza93

File: app/settings/profile.coffee (hunk 1 above)
Comment (ncino-cpezza) [reply app/settings/profile.coffee]: updated to be a removeProfile function @npezza93

File: app/settings/profile.coffee (hunk 2 above)
Comment (ncino-cpezza) [reply app/settings/profile.coffee]: Update to proper x usage

File: app/settings/profiles.coffee [hunk 4]
```diff
@@ -47,6 +48,22 @@ class Profiles extends React.Component
...
+    Object.assign(tempState, settings)
+    if tempState.activeProfile == id
+      profileId = Object.keys(tempState.profiles)[0]
+      tempState.activeProfile = Object.keys(tempState.profiles)[0]
+
+    delete tempState.profiles[id]
+    @_configFile.write(tempState)
+
+    @setState(
+      activeProfile: profileId, profiles: Object.values(settings.profiles)
```
Comment (npezza93) [inline app/settings/profiles.coffee]: Just use tempState.activeProfile and remove the temp variable profileId

File: app/settings/settings.css [hunk 5]
```diff
@@ -97,6 +97,24 @@ input,
...
+  position: absolute;
+  left: 15px;
+  bottom: 17px;
```
Comment (npezza93) [inline app/settings/settings.css]: 16px

Comment (techgurupezza) [reply app/settings/settings.css]: for left or bottom? @npezza93

File: app/settings/profiles.coffee (hunk 4 above)
Comment (techgurupezza) [reply app/settings/profiles.coffee]: after you write the tempState is null

File: app/settings/settings.css (hunk 5 above)
Comment (npezza93) [reply app/settings/settings.css]: bottom

File: app/settings/settings.css (hunk 3 above)
Comment (techgurupezza) [reply app/settings/settings.css]: Moved to left and added hover state

File: app/settings/profile.coffee (hunk 2 above)
Comment (techgurupezza) [reply app/settings/profile.coffee]: Update to proper x usage

File: app/settings/profile.coffee (hunk 1 above)
Comment (techgurupezza) [reply app/settings/profile.coffee]: updated to be a removeProfile function

File: app/settings/settings.css (hunk 5 above)
Comment (techgurupezza) [reply app/settings/settings.css]: ok updated here

File: app/settings/profiles.coffee (hunk 4 above)
Comment (techgurupezza) [reply app/settings/profiles.coffee]: stand corrected, after moving to `profiles.coffee` this was allowed. so corrected it in new commit @npezza93

File: app/settings/profiles.coffee [hunk 6]
```diff
@@ -47,6 +48,21 @@ class Profiles extends React.Component
 
     @setState(activeProfile: id)
 
+  removeProfile: (id) ->
```
Comment (npezza93) [inline app/settings/profiles.coffee]:
```  removeProfile: (id) ->
    tempState = {}

    Object.assign(tempState, @settings())
    if tempState.activeProfile == id
      tempState.activeProfile = parseInt(Object.keys(tempState.profiles)[0])

    delete tempState.profiles[id]
    @_configFile.write(tempState)

    @setState(
      activeProfile: @settings().activeProfile
      profiles: Object.values(@settings().profiles)
    )
```

Comment (techgurupezza) [reply app/settings/profiles.coffee]: cool thanks updated!

**Pull request merged**